kuwala/core/canvas/src/data/state/models/CanvasModel.js

```diff
@@ -151,6 +178,10 @@ const CanvasModel = {
                 sourceDTO = sourceElement.data.transformationBlock;
                 sourceDTO.connectedTargetNodeIds = sourceDTO.connectedTargetNodeIds.filter((el) => el !== target);
                 updateTransformationBlock({ addNode, elements, setElements, updatedBlock: sourceDTO });
+            } else if (sourceElement.type === EXPORT_BLOCK) {
+                sourceDTO = sourceElement.data.exportBlock;
+                sourceDTO.connectedTargetNodeIds = sourceDTO.connectedTargetNodeIds.filter((el) => el !== target);
+                updateExportBlock({ addNode, elements, setElements, updatedBlock: sourceDTO });
```

Comment:
This action can be moved outside of all the blocks and only conditionally set the `sourceDTO`.